content/post/0-to-60-analytics-stack.md

```diff
@@ -0,0 +1,79 @@
+company that historically lacked a comprehensive data strategy. You sold the vision and also put together a roadmap based on your experience and current best practices. What could go wrong (and who could argue that refactoring from a star to snowflake schema is an important company-wide goal)?
+
+The reality is that, like any other function, analytics and data exist within the
+context of an organization - an organization that has competing priorities,
+politics, and limited resourcing. I wrote this article to help you combine pragmatism with best practices and prepare for the discussions with engineering, finance, and possibly exec that you did not realize you needed to have. 
+
+Over the past 5 years, analytics technology has coalesced around three components: a data-moving tool (possibly referred by EL or ETL), a data warehouse to store beforementioned data, and a BI layer to analyze the data. As your company grows, you will also realize you need a centralized business-logic layer, as well as an org plan for empowering teams to use analytics when the company is too big for you to personally know everyone.
+
+## ETL Tool:
+
+- Some solutions you might consider: Stitch, Fivetran, tray.io
+
+For finance, most data-movers costs far less than a full-time Data Engineer, so buying this component saves the company money from paying an engineer to do (unsatisfying) data work. However, some providers bill per number of rows synced, while some bill per number of unique data sources. You may find that finance prefers to the billing structure of your second choice much more than your first choice provider! Be prepared to either convince finance that you can keep variable costs under control, or convince the sales rep to customize your contract.
+
+From the engineering side, you may get “Don't waste money - we could do this ourselves, it is so easy”. Be prepared to ask if your well-intentioned teammate:
+
+1. Has actually implemented and maintained a data pipeline for several years
+2. Is personally volunteering to do so, for you, in a timely manner
+3. Is excited to be on-call 24/7 to fix issues.
+
+
+## Analytics Database:
+
+- Some solutions you might consider: Postgres, Redshift, Snowflake, Google Cloud
+
+There are more posts comparing analytics databases than there are rows *in* my database, so I will spare details and instead discuss a useful framework for choosing open-source vs. paid. 
+It is tempting to spin up a open-source database and use it for analytics, but you should carefully evaluate if the benefit of cost savings is worth the switching costs you will incur later. 
+
+In my case, I balked at the price of snowflake and tried to use a postgres instance for analytics because we had a small amount of data. This was dumb, because the savings of (snowflake – postgres) * (time before we needed to switch to snowflake) is less than the cost of spending time trying to diagnose random postgres issues once a week. Don't undervalue your own time!
```

Comment:
Maybe say "In hindsight this was a poor decision" instead of "This was dumb". A lot of folks out there might have made the same decision so don't want to be overly critical.